pkg/typesystem/typesystem.go

```diff
@@ -186,6 +189,7 @@ func New(model *openfgav1.AuthorizationModel) *TypeSystem {
 		schemaVersion:   model.GetSchemaVersion(),
 		typeDefinitions: tds,
 		relations:       relations,
+		conditions:      model.GetConditions(),
```

Comment:
```suggestion
		conditions := make(map[string]*CompiledCondition, len(model.Conditions))
                for _, c := range model.conditions {
                         conditions[c.name] = condition.Compile(c.expression)
                  }
```

Reply:
gotta think about this a bit more. `condition.Compile()` can currently return an error which wouldn't make sense in the context of `typesystem.New()`

Reply:
It's assumed that by the time you're constructing a TypeSystem using `typesystem.New` that the underlying model has been validated. So if we see an error when compiling a condition when constructing a TypeSystem, then we can panic with the error because a serious invariant has not been met in the `WriteAuthorizationModel` API. 

See `typesystem.NewAndValidate(..)` instead.